x-pack/plugins/security_solution/public/flyout/README.md

```diff
@@ -1,44 +1,61 @@
+The Security Solution plugin aims at having a single instance of the expandable flyout. That instance can display as many panels as we need. This folder hosts all the panels that are can be displayed in the Security Solution flyout. Panels can be differentiated as to be displayed in different sections of the expandable flyout (right, left or preview), but ultimately, nothing prevents us from displaying a panel in any section we want.
 
-> Remember to add any new panels to the `index.tsx` at the root of the `flyout` folder. These are passed to the `@kbn/expandable-flyout` package as `registeredPanels`.
+> Remember to add any new panels to the `index.tsx` at the root of the [flyout folder](https://github.com/elastic/kibana/tree/main/x-pack/plugins/security_solution/public/flyout). These are passed to the `@kbn/expandable-flyout` package as `registeredPanels`. Failing to do so will result in the panel not being rendered.
 
 ## Notes
 
-At the moment, we only have a single expandable flyout for the Security Solution plugin. This flyout will be used for all documents (signals, events, indicators, assets and findings). We're using a set of generic right/left/preview panels, hence the following folder structure:
-
+The structure of the `flyout` folder is intended to work as follows:
+- top level folders refer to the _type_ of flyout (for example alert details, event details, user, host, rule, cases...) and would contain all the panels for that flyout _type_.
+- each of these top level folders can be organized the way you want, but we recommend following a similar structure to the one we have for the `document_details` flyout type:
 ```
-flyout
-│   index.tsx
-│   README.md
-│
-└───right
-└───left
-└───preview
+document_details
+└─── right
+└─── left
+└─── preview
+└─── shared
 ```
+Where the `right`, `left` and `preview` folders correspond to the panels displayed in the right, left and preview sections respectively. The `shared` folder contains any shared components/hooks/services/helpers that are used within the other folders. 
 
-If different right, left or preview panels are needed, we should refactor the folder structure as follows:
-
+The `flyout` folder structure should look like this:
 ```
 flyout
 │   index.tsx
+│   jest.config.js
 │   README.md    
 │
-└───documents
-│   └───right
-│   └───left
-│   └───preview
+└─── document_details
+│   └─── right
+│   └─── left
+│   └─── preview
 │
-└───new_type
-│   └───right
-│   └───left
-│   └───preview
+└─── event_details
+│   └─── right
+│   └─── left
+│   └─── preview
 │
-└───other_new_type
-    └───right
-    └───left
-    └───preview
+└─── new_type
+│   └─── right
+│   └─── preview
+│
+└─── other_new_type
+│   └─── right
+│   └─── left
+│
+└─── ...
+│
+└─── shared
+    └─── components
 ```
+
+Where the top level `shared` folder contains all the components/hooks/services/helpers that are used accross multiple flyout types.
```

Comment:
nit: consider moving this section before line 25. i think it helps to layout there are 2 types of top level folders: 1) type folders, 2) shared folder, then show suggested structure at the end. wdyt?

Reply:
I moved this section up and rewrote the bullet points, let me know if they make more sense